Added Managing cluster configuration topic

gaurav-nelson opened the pull request:
Based on https://help.stackrox.com/docs/manage-cluster-configuration/

Preview at https://openshift-docs-git-cluster-config-gnelson.vercel.app/openshift-acs/master/operating/manage-cluster-configuration.html

pneedle-rh commented on modules/unused-kubernetes-roles.adoc:
```diff
@@ -0,0 +1,19 @@
+The information appears in a panel on the right.
+. To find roles not granted to anyone, select the *Users & Groups* column header.
+Then, while holding the kbd:[Shift] key, select the *Service Account* column header.
+The list gets sorted to show roles that are not granted to any users, groups, or service accounts.
```
You could minimise this sentence as follows:

"The list shows the roles that are not granted to any users, groups, or service accounts."

pneedle-rh commented on modules/view-kubernetes-secrets.adoc:
```diff
@@ -0,0 +1,19 @@
+= Viewing Kubernetes secrets
+
+[role="_abstract"]
+View Kubernetes secrets in use in your environment and identify deployments using those secrets.
```
Does deployments refer to clusters? If so, you could change "deployments" to "clusters" in this line to make it more explicit.

pneedle-rh commented on modules/view-kubernetes-secrets.adoc:
```diff
@@ -0,0 +1,19 @@
+.Procedure
+. Navigate to the RHACS portal and click *Configuration Management* from the left-hand navigation menu.
+. On the *Secrets Most Used Across Deployments* widget, select *View All*.
+Alternatively, select *Application & Infrastructure* -> *Secrets* from the *Configuration Management* view header.
```
You could consider only including one of the methods if both routes navigate you to the same view, to minimise the step.

pneedle-rh commented on modules/view-kubernetes-secrets.adoc:
```diff
@@ -0,0 +1,19 @@
+Alternatively, select *Application & Infrastructure* -> *Secrets* from the *Configuration Management* view header.
+The *Secrets* view displays a list of Kubernetes secrets.
+. Select a row to view more details.
+The information appears in a panel on the right.
```
You could potentially remove this line because this would be self-explanatory in the UI and it might also change position over time.

pneedle-rh commented on modules/find-policy-violations.adoc:
```diff
@@ -0,0 +1,30 @@
+= Finding policy violations
+
+[role="_abstract"]
+The *Policy Violations by Severity* widget in the *Configuration Management* view displays policy violations in a sunburst chart.
```
In a future PR you could consider moving this conceptual content to a concept module that is included before this procedure module.

pneedle-rh commented on modules/find-policy-violations.adoc:
```diff
@@ -0,0 +1,30 @@
+Each level of the chart is represented by one ring or circle.
+
+* The innermost circle represents the total number of violations.
+* The next ring represents policy categories (*Low*, *Medium*, *High*, and *Critical*).
```
You could remove the parentheses from this line as follows:

"* The next ring represents the **Low**, **Medium**, **High**, and **Critical** policy categories."

pneedle-rh commented on modules/find-policy-violations.adoc:
```diff
@@ -0,0 +1,30 @@
+.Procedure
+. Navigate to the RHACS portal and click *Configuration Management* from the left-hand navigation menu.
+. On the *Secrets Most Used Across Deployments* widget, move your mouse over the sunburst chart to view details about policy violations.
+. Select *_n_ rated as high* (where _n_ is a number) to view detailed information about high-priority policy violations.
```
You could remove the parentheses from this sentence as follows:

". Select **_n_ rated as high**, where _n_ is a number, to view..."

pneedle-rh commented on modules/find-policy-violations.adoc:
```diff
@@ -0,0 +1,30 @@
+. Navigate to the RHACS portal and click *Configuration Management* from the left-hand navigation menu.
+. On the *Secrets Most Used Across Deployments* widget, move your mouse over the sunburst chart to view details about policy violations.
+. Select *_n_ rated as high* (where _n_ is a number) to view detailed information about high-priority policy violations.
+The *Policies* view displays a list of policy violations filtered based on the selected category.
```
You could remove "based" from this sentence.

pneedle-rh commented on modules/find-policy-violations.adoc:
```diff
@@ -0,0 +1,30 @@
+. Select a row to view more details, including policy description, remediation, deployments with violations, and more.
+The details are visible in a panel.
+. The *Policy Findings* section in the information panel lists deployments where these violations occurred.
+. Select a deployment under the *Policy Findings* section to view related details including Kubernetes labels, annotations, service account, and violation comments and tags.
```
You could potentially replace "deployment" with "cluster" to be more explicit, unless deployment is referring to something else. The same applies to other references to deployment in the PR.

gaurav-nelson replied:
No its not a cluster, its information about the Kubernetes deployment object. This is correct.

pneedle-rh commented on modules/find-policy-violations.adoc:
```diff
@@ -0,0 +1,30 @@
+. The *Policy Findings* section in the information panel lists deployments where these violations occurred.
+. Select a deployment under the *Policy Findings* section to view related details including Kubernetes labels, annotations, service account, and violation comments and tags.
+
+Use the detailed information to plan a remediation for violations.
```
You could potentially change "Use" to "You can use".

pneedle-rh commented on modules/find-failing-cis-controls.adoc:
```diff
@@ -0,0 +1,28 @@
+= Finding failing CIS controls
+
+[role="_abstract"]
+Similar to the *Policy Violations* sunburst chart in the *Configuration Management* view, the CIS controls widget provides information about failing CIS controls.
```
You could provide the full name of the CIS acronym before the first usage in this module. For example:

"Similar to the *Policy Violations* sunburst chart in the *Configuration Management* view, the Center for Information Security (CIS) controls widget provides"

The same applies to other modules that use the acronym. Those modules might be used in other assemblies that do not provide the full name in preceding content.

pneedle-rh replied:
Is the "CIS controls" widget a UI object. If so, you could make the name bold. The same applies to other references in the PR, but I won't mention them all in this review.

pneedle-rh replied:
In a future PR, you could consider moving this conceptual content to a concept module that is included before this procedure module.

pneedle-rh commented on modules/find-failing-cis-controls.adoc:
```diff
@@ -0,0 +1,28 @@
+. Select *CIS Docker v1.2.0* on the CIS controls widget header.
+Use this to switch between CIS Docker and Kubernetes controls.
+. Hover over the sunburst chart to view details about failing controls.
+. Select *_n_ controls failing* (where _n_ is a number) to view detailed information about failing controls.
```
You could remove the parentheses from this sentence as follows:

". Select **_n_ controls failing**, where _n_ is a number, to view..."

pneedle-rh commented on modules/find-failing-cis-controls.adoc:
```diff
@@ -0,0 +1,28 @@
+. Select *_n_ controls failing* (where _n_ is a number) to view detailed information about failing controls.
+The *Controls* view displays a list of failing controls filtered based on the compliance state.
+. Select a row to view more details, including control descriptions and nodes where the controls are failing.
+The details are visible in a panel.
```
You could potentially remove this line because it is self-explanatory in the UI.

pneedle-rh commented on modules/find-failing-cis-controls.adoc:
```diff
@@ -0,0 +1,28 @@
+. The *Control Findings* section in the information panel lists nodes where the controls are failing.
+Select a row to view more details, including Kubernetes labels, annotations, and other metadata.
+
+Use the detailed information to focus your attention on a subset of nodes, industry standards, or failing controls, and assess, check, and report on the compliance status of your containerized infrastructure.
```
You could change *Use" to "You can use".

You could also remove "your attention" from this sentence.

pneedle-rh replied:
You could split this sentence into two, to facilitate localisation. For example:

"You can use the detailed information to focus on a subset of nodes, industry standards, or failing controls. You can also assess, check, and report on the compliance status of your containerized infrastructure."

pneedle-rh commented on modules/kubernetes-role-assignment.adoc:
```diff
@@ -0,0 +1,16 @@
+Use the *Configuration Management* view to get information about which Kubernetes roles are assigned to which users and groups.
+
+.Procedure
+. Navigate to the RHACS portal and click *Configuration Management* from the left-hand navigation menu.
```
After reading through a few more of the PRs, I realise that the product name "Red Hat Advanced Cluster Security for Kubernetes" is always referred to in full in the docs and the portal for the product is always referred to as "the RHACS portal".

In hindsight, you could consider referring to the full name of the portal the first time you use it and then use the acronym for further references, in each module.

Please see [this related conversation in another PR](https://github.com/openshift/openshift-docs/pull/34199/files#r669543422).

gaurav-nelson replied:
For this PR I have added the acronym in the assembly. I'll updated all assemblies to always have that acronym appear along with first mention of the name so that I don't have to include full name in all modules.

gaurav-nelson commented on modules/configuration-management-view.adoc:
```diff
@@ -0,0 +1,28 @@
+Similar to the *Dashboard*, it displays some useful widgets.
+
+[discrete]
+== Widgets
```
That's a long title. Instead I think I can remove the headings.

It can just be the description.

gaurav-nelson commented on modules/configuration-management-view.adoc:
```diff
@@ -0,0 +1,28 @@
+
+[discrete]
+== Header
+The *Configuration Management* view header shows you the number of policies and CIS controls and allows you to switch between different entities.
```
Sure, we used `view header` as a UI component name in our previous docs.

gaurav-nelson commented on operating/manage-cluster-configuration.adoc:
```diff
@@ -0,0 +1,49 @@
+
+
+[id="using-comments-and-tags"]
+== Using comments and tags
```
I've moved the comments and tagging section in the finding policy section.

gaurav-nelson commented:
@pneedle-rh please review the updates. I've addressed all comments except changing the introduction of the assembly. I like it as it is. But let me know if it is absolutely required to change the introductory text before this can be merged, and I can change it.

pneedle-rh commented:
This LGTM!
